core-library/src/main/java/org/silverpeas/core/admin/user/GroupManager.java

```diff
@@ -23,50 +23,72 @@
-import org.silverpeas.core.admin.persistence.AdminPersistenceException;
-import org.silverpeas.core.admin.persistence.GroupRow;
+import org.silverpeas.core.admin.persistence.OrganizationSchema;
+import org.silverpeas.core.admin.persistence.SpaceUserRoleRow;
+import org.silverpeas.core.admin.persistence.SpaceUserRoleTable;
+import org.silverpeas.core.admin.persistence.UserRoleRow;
+import org.silverpeas.core.admin.persistence.UserRoleTable;
 import org.silverpeas.core.admin.service.AdminException;
 import org.silverpeas.core.admin.user.constant.UserState;
 import org.silverpeas.core.admin.user.dao.GroupDAO;
 import org.silverpeas.core.admin.user.dao.GroupSearchCriteriaForDAO;
 import org.silverpeas.core.admin.user.dao.SearchCriteriaDAOFactory;
 import org.silverpeas.core.admin.user.dao.UserDAO;
 import org.silverpeas.core.admin.user.dao.UserSearchCriteriaForDAO;
-import org.silverpeas.core.admin.user.model.AdminGroupInst;
-import org.silverpeas.core.admin.user.model.GroupDetail;
 import org.silverpeas.core.admin.user.model.GroupDetail;
 import org.silverpeas.core.admin.user.notification.GroupEventNotifier;
 import org.silverpeas.core.notification.system.ResourceEvent;
 import org.silverpeas.core.persistence.jdbc.DBUtil;
-import org.silverpeas.core.util.ArrayUtil;
 import org.silverpeas.core.util.ListSlice;
+import org.silverpeas.core.util.ServiceProvider;
 import org.silverpeas.core.util.StringUtil;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
+import javax.transaction.Transactional;
 import java.sql.Connection;
 import java.sql.SQLException;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import static org.silverpeas.core.SilverpeasExceptionMessages.*;
 
 @Singleton
+@Transactional(Transactional.TxType.MANDATORY)
 public class GroupManager {
 
+  public static final String GROUP = "group";
+  public static final String GROUP_MANAGER_GET_GROUPS_OF_DOMAIN =
+      "GroupManager.getGroupsOfDomain()";
+  public static final String GROUP_MANAGER_ADD_GROUP = "GroupManager.addGroup()";
+  public static final String GROUP_MANAGER_UPDATE_GROUP = "GroupManager.updateGroup()";
+  public static final String IN_GROUP = "in group ";
+  public static final String GROUP_MANAGER_DELETE_GROUP = "GroupManager.deleteGroup()";
+  public static final String REMOVING_MESSAGE = "Suppression de ";
+  public static final String IN_SILVERPEAS_MESSAGE = " dans la base";
   @Inject
   private GroupDAO groupDao;
   @Inject
   private UserDAO userDao;
   @Inject
   private GroupEventNotifier notifier;
+  @Inject
+  private OrganizationSchema organizationSchema;
+  @Inject
+  private DomainDriverManager domainDriverManager;
+
+  public static GroupManager get() {
+    return ServiceProvider.getService(GroupManager.class);
+  }
 
   /**
    * Constructor
    */
-  private GroupManager() {
+  protected GroupManager() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)